### PR TITLE
Make Glue script --profile and --junit work

### PR DESCRIPTION
**Pull request by lilyeyes**
Fix Glue script --profile and --junit features does not work at the same time 
Add e2e test for junit and profile

Related ticket: [TEAM-9475](https://jira.suse.com/browse/TEAM-9475) - Glue script --profile and --junit features does not work at the same time

VRS: 
(The `deploy-qesap_exec_ansible.log.txt` contains lines like: `DEBUG    OUTPUT: Sunday 07 July 2024  23:39:58 -0400 (0:00:02.735)       0:00:02.741 *********** `)
15-SP3: https://openqaworker15.qa.suse.cz/tests/288381
12-sp5: https://openqaworker15.qa.suse.cz/tests/288422

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

**Comment by badboywj** [comment]
LGTM, thanks

File: scripts/qesap/test/e2e/test.sh [hunk 1]
```diff
@@ -288,3 +288,24 @@ ansible_logs_number=$(find . -type f -name "ansible.*.log.txt" | wc -l)
...
+
+test_step "[test_4.yaml] Run Ansible with --junit"
+find . -type f -name "sambuconero*.xml" -delete || echo "Nothing to delete"
+qesap.py --verbose -b ${QESAPROOT} -c test_4.yaml ansible --junit . || test_die "test_4.yaml fail on ansible"
+junit_logs_number=$(find . -type f -name "sambuconero*.xml" | wc -l)
+[[ $junit_logs_number -eq 1 ]] || test_die "ansible JUNIT reports should be 1 files but are ${junit_logs_number}"
+find . -type f -name "sambuconero*.xml" -delete
+rm ansible.*.log.txt
+
+test_step "[test_4.yaml] Run Ansible with --profile"
```
Comment (mpagot) [inline scripts/qesap/test/e2e/test.sh:300]: this test can be improved. It is fine but it does not check is profile info are in the output or not. Please incorporate test implementation from https://github.com/SUSE/qe-sap-deployment/pull/254/files#diff-211ace378c4b9937437dae88a2a8e6f50efb2315618a3f77a124410274d8f36dR305

It is also fine you just look at it and provide your own implementation

File: scripts/qesap/test/e2e/test.sh [hunk 2]
```diff
@@ -288,3 +288,24 @@ ansible_logs_number=$(find . -type f -name "ansible.*.log.txt" | wc -l)
...
+qesap.py --verbose -b ${QESAPROOT} -c test_4.yaml ansible --profile || test_die "test_4.yaml fail on ansible"
+find . -type f -name "sambuconero*.xml" -delete
+rm ansible.*.log.txt
+
+test_step "[test_4.yaml] Run Ansible with --profile and --junit"
```
Comment (mpagot) [inline scripts/qesap/test/e2e/test.sh:305]: the same for this last test, junit part is ok but https://github.com/SUSE/qe-sap-deployment/pull/254 has improvement for the profile one

Comment (mpagot) [reply scripts/qesap/test/e2e/test.sh:305]: For example I think that this implementation of the e2e test is not able to spot the issue that this PR is fixing, this test does not fails without your fix in the code and it should.

Comment (lilyeyes) [reply scripts/qesap/test/e2e/test.sh:305]: I will change the code later, thanks for the comments.

File: scripts/qesap/test/e2e/test.sh (hunk 1 above)
Comment (lilyeyes) [reply scripts/qesap/test/e2e/test.sh]: Done. Thanks for the comments.

File: scripts/qesap/test/e2e/test.sh (hunk 2 above)
Comment (lilyeyes) [reply scripts/qesap/test/e2e/test.sh]: Done. Thanks for the comments.

**Review by mpagot — APPROVED** [review]
LGTM

**Pull request merged**